tb_plugin/torch_tb_profiler/profiler/loader.py

```diff
@@ -42,19 +32,62 @@ def _parse(self):
             worker = match.group(1)
             workers.append((worker, path))
 
+        barrier = Barrier(len(workers) + 1)
         for worker, path in sorted(workers):
-            try:
-                data = RunProfileData.parse(self.run.run_dir, worker, path, self.caches)
-                self.run.profiles[worker] = data
-            except Exception as ex:
+            p = Process(target=self._process_data, args=(worker, path, barrier))
+            p.start()
+
+        logger.info("starting all processing")
+        # since there is one queue, its data must be read before join.
+        # https://stackoverflow.com/questions/31665328/python-3-multiprocessing-queue-deadlock-when-calling-join-before-the-queue-is-em
+        barrier.wait()
```

Comment:
How about just counting the queue.get() and exiting loop on reaching "workers"? Then Barrier could be removed.

Reply:
We need handle some worker process return nothing.